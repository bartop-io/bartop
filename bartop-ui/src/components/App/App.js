import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import persistState from 'redux-localstorage';
import { Router, Route, Switch, Redirect } from 'react-router-dom';
import { injectGlobal } from 'styled-components';

import callApiMiddleware from '../../middleware/call-api';
import rootReducer from '../../ducks';
import auth from '../../singletons/authentication';
import history from '../../singletons/history';
import Landing from '../Landing/Landing';
import Auth from '../Auth/Auth';
import NotFound from '../NotFound/NotFound';

// persist all of our auth & user state to local storage
const localStorageEnhancer = persistState(['authentication', 'user'], {
  key: 'bartop'
});

const enhancer = composeWithDevTools(
  applyMiddleware(thunk, callApiMiddleware),
  localStorageEnhancer
);

const store = createStore(
  rootReducer,
  undefined, // optional preloaded state
  enhancer // compose to combine enhancers for middlewares like thunk and other enhancers like dev tools
);

export default class App extends React.Component {
  render() {
    // styled components helper for adding styles to global dom elements like body
    injectGlobal`
      html,
      body,
      #root {
        width: 100%;
        height: 100%;
        margin: 0;
        padding: 0;
        font-family: sans-serif;
      }
    `;
    return (
      <Provider store={store}>
        <Router history={history}>
          <Switch>
            <Redirect exact from="/" to="/landing" />
            <Route path="/landing" component={Landing} />
            <Route
              path="/auth"
              render={({ match, history }) => (
                <Auth match={match} history={history} auth={auth} />
              )}
            />
            <Route component={NotFound} />
          </Switch>
        </Router>
      </Provider>
    );
  }
}
