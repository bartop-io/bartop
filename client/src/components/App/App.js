import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { Router, Route } from 'react-router-dom';
import { injectGlobal } from 'styled-components';

import rootReducer from '../../ducks';
import history from '../../history';
import Auth from '../../authentication/authentication';
import Landing from '../Landing/Landing';

// if in development, enable redux dev tools
const enhancer =
  process.env.NODE_ENV === 'development'
    ? window.__REDUX_DEVTOOLS_EXTENSION__ &&
      window.__REDUX_DEVTOOLS_EXTENSION__()
    : undefined;
// second argument is optional preloaded state so we pass undefined
const store = createStore(rootReducer, undefined, enhancer);

const auth = new Auth();

export default class App extends React.Component {
  // throw auth in context so any child components can use it if needed
  getChildContext() {
    return { auth };
  }
  render() {
    // styled components helper for adding styles to global dom elements like body
    injectGlobal`
      body {
        margin: 0;
        padding: 0;
        font-family: sans-serif;
      }
    `;
    return (
      <Provider store={store}>
        <Router history={history}>
          <Route path="/" exact component={Landing} />
        </Router>
      </Provider>
    );
  }
}

App.childContextTypes = {
  auth: PropTypes.object
};
