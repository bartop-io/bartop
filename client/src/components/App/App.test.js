import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

jest.mock('redux-localstorage');
jest.mock('redux');

it('renders', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.exists()).toBe(true);
});
