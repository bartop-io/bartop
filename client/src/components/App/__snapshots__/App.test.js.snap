// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <div>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        exact={true}
        path="/callback"
        render={[Function]}
      />
    </div>
  </Router>
</Provider>
`;
